clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpZeebeFuture.java

```diff
@@ -56,6 +55,20 @@ public boolean cancel(final boolean mayInterruptIfRunning) {
     return super.cancel(mayInterruptIfRunning);
   }
 
+  @Override
+  public RespT join(final long timeout, final TimeUnit unit) {
+    try {
+      return super.get(timeout, unit);
+    } catch (final ExecutionException e) {
+      throw new ClientException(e);
```

Comment:
🔧 Don't we need to unwrap the exception here as well using `unwrapExecutionException`?

Reply:
That is also what we do in the `ZeebeClientFutureImpl` ([src](https://github.com/camunda/camunda/blob/1376ef75260543ddb917e165f5c02d5a4c331f73/clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeClientFutureImpl.java#L74)).
I'll adjust this, add a test case, and merge this PR if it succeeds.

Reply:
```suggestion
      throw unwrapExecutionException(e);
```